p2p/config/config.go

```diff
@@ -27,72 +25,48 @@ func duration(duration string) (dur time.Duration) {
+	NewNode           bool            `mapstructure:"new-node"`
+	DialTimeout       time.Duration   `mapstructure:"dial-timeout"`
+	ConnKeepAlive     time.Duration   `mapstructure:"conn-keepalive"`
+	NetworkID         int8            `mapstructure:"network-id"`
+	DiscoveryConfig   DiscoveryConfig `mapstructure:"dht"`
+	BufferSize        int             `mapstructure:"buffer-size"`
+	Gossip            bool            `mapstructure:"gossip"`
+	RandomConnections int             `mapstructure:"randcon"`
 }
 
-// SwarmConfig specifies swarm config params.
-type SwarmConfig struct {
-	Gossip                 bool     `mapstructure:"gossip"`
+// DiscoveryConfig specifies swarm config params.
+type DiscoveryConfig struct {
 	Bootstrap              bool     `mapstructure:"bootstrap"`
 	RoutingTableBucketSize int      `mapstructure:"bucketsize"`
 	RoutingTableAlpha      int      `mapstructure:"alpha"`
-	RandomConnections      int      `mapstructure:"randcon"`
 	BootstrapNodes         []string `mapstructure:"bootnodes"`
 }
 
-// TimeConfig specifies the timesync params for ntp.
-type TimeConfig struct {
-	MaxAllowedDrift       time.Duration `mapstructure:"max-allowed-time-drift"`
-	NtpQueries            int           `mapstructure:"ntp-queries"`
-	DefaultTimeoutLatency time.Duration `mapstructure:"default-timeout-latency"`
-	RefreshNtpInterval    time.Duration `mapstructure:"refresh-ntp-interval"`
-}
-
 // DefaultConfig deines the default p2p configuration
 func DefaultConfig() Config {
 
-	// TimeConfigValues defines default values for all time and ntp related params.
-	var TimeConfigValues = TimeConfig{
-		MaxAllowedDrift:       duration("10s"),
-		NtpQueries:            5,
-		DefaultTimeoutLatency: duration("10s"),
-		RefreshNtpInterval:    duration("30m"),
-	}
-
-	// SwarmConfigValues defines default values for swarm config params.
-	var SwarmConfigValues = SwarmConfig{
-		Gossip:                 false,
+	// DHTConfigValues defines default values for swarm config params.
+	var DHTConfigValues = DiscoveryConfig{
 		Bootstrap:              false,
 		RoutingTableBucketSize: 20,
 		RoutingTableAlpha:      3,
-		RandomConnections:      5,
-		BootstrapNodes:         []string{ // these should be the spacemesh foundation bootstrap nodes
+		BootstrapNodes: []string{ // these should be the spacemesh foundation bootstrap nodes
 		},
 	}
 
 	return Config{
-		SecurityParam:   20,
-		FastSync:        true,
-		TCPPort:         7513,
-		NodeID:          "",
-		NewNode:         false,
-		DialTimeout:     duration("1m"),
-		ConnKeepAlive:   duration("48h"),
-		NetworkID:       TestNet,
-		ResponseTimeout: duration("15s"),
-		SwarmConfig:     SwarmConfigValues,
-		TimeConfig:      TimeConfigValues,
-        BufferSize:      100,
+		TCPPort:           7513,
+		NodeID:            "",
+		NewNode:           false,
+		DialTimeout:       duration("1m"),
+		ConnKeepAlive:     duration("48h"),
+		NetworkID:         TestNet,
+		DiscoveryConfig:   DHTConfigValues,
```

Comment:
replace `DHTConfigValues` with a `DefaultConfig()` call

Reply:
remove `***Values` completely from out system.

Reply:
> replace `DHTConfigValues` with a `DefaultConfig()` call

Done in e62de719aeb2807044b93420b6dbc8b0926ced59.

Reply:
> remove `***Values` completely from out system.

There are several places with `***Values`. Can we remove them with another pull request? This one is a little big.